pkg/s3wrapper/client.go

```diff
@@ -414,7 +420,14 @@ func (c *S3Client) ListObjectsByPrefix(ctx context.Context, prefix string) ([]st
 		return nil, err
 	}
 	for _, key := range resp.Contents {
-		objects = append(objects, *key.Key)
+		// We need to find the metadata for each object
+		headObjectOutput, err := c.client.HeadObject(&s3.HeadObjectInput{Bucket: aws.String(c.cfg.S3Bucket), Key: key.Key})
+		if err != nil {
+			err = errors.Wrapf(err, "Failed to retrieve metadata for %s", *key.Key)
+			log.Error(err)
+			return nil, err
+		}
+		objects = append(objects, ObjectInfo{Path: *key.Key, Metadata: headObjectOutput.Metadata})
```

Comment:
I mean we could simplify this by doing two things.

1: As you suggest, we could set aws.Config.LowerCaseHeaderMaps to true
2: In addition to this, we could ensure that the S3 client (and correspondingly the filesystem client) reduce key names (for user keys) to lowercase. Meaning that they become case insensitive.

I don't see a reason to maintain case sensitivity so I think this would work?

WDYT @pastequo ?

Reply:
I think universally enforcing that all metadata key writes are interpreted as lowercase values is the way to go.

So if the user supplies the key `Some-AttributeKey` it would be interpreted as `some-attributekey` 

I think we can achieve this by using `strings.ToLower` in the filesystem handler and using the `aws.Config.LowerCaseHeaderMaps` setting in the S3 handler.

I will implement this.

Reply:
I think that S3 metadata are effectively headers for each file and are a map of strings, I think that potentially there are headers that are not metadata but these are automatically filtered out already.

In terms of sanitization, we are dealing with string data and nothing in the code (at this point in time) makes an attempt to interpret metadata values as anything else. I don't think there is additional sanitization or filtering needed, unless I have missed something?

Can you provide an example of what you mean?

Reply:
Already taken into account then :)
I perform a strings.ToLower on the key for the filesystem implementation. `aws.Config.LowerCaseHeaderMaps` does the same job on the S3 side